cpp/src/arrow/filesystem/filesystem.h

```diff
@@ -36,8 +36,15 @@
 #include "arrow/util/windows_fixup.h"
 
 namespace arrow {
+
+namespace internal {
+class Uri;
+}
+
 namespace fs {
 
+using arrow::internal::Uri;
```

Comment:
Since this is going to be exposed in a non-internal API, perhaps we want to move `Uri` into `arrow::util` instead?

Reply:
Will do